gogoproto/gogo.proto

```diff
@@ -103,5 +103,12 @@ extend google.protobuf.FieldOptions {
   optional string jsontag = 65005;
   optional string moretags = 65006;
   optional string casttype = 65007;
+	optional FieldValidator validator = 65008;
 }
 
+message FieldValidator {
+	optional string regex = 1;
+	optional int64 int_gt = 2;
+	optional int64 int_lt = 3;
+	optional bool msg_exists = 4;
```

Comment:
Unfortunately, you can't do that :( In `proto3` the fields are always defaults (`""` for string or `0` for int) and you can't differentiate between them being present or not. 
The only one you can is submessages.
Also, `required` is a proto keyword :(